nifi-nar-bundles/nifi-extension-utils/nifi-database-utils/src/test/java/org/apache/nifi/util/db/TestJdbcHugeStream.java

```diff
@@ -53,21 +52,18 @@
  */
 public class TestJdbcHugeStream {
 
-    @Rule
-    public TemporaryFolder folder = new TemporaryFolder();
-
-    @BeforeClass
+    @BeforeAll
     public static void setup() {
         System.setProperty("derby.stream.error.file", "target/derby.log");
     }
 
     @Test
     public void readSend2StreamHuge_FileBased() throws ClassNotFoundException, SQLException, IOException {
-
-        // remove previous test database, if any
-        folder.delete();
-
-        try (final Connection con = createConnection(folder.getRoot().getAbsolutePath())) {
+        String path = Files.createTempDirectory(String.valueOf(System.currentTimeMillis()))
+                .resolve("db")
+                .toFile()
+                .getAbsolutePath();
```

Comment:
This test appears to have issues similar to those in `DBCPServiceTest`. See the approach in that test class where `derby.log` is placed in one location, and the database is placed in another empty directory, which is cleared after test completion.

Reply:
This test also needs a method for clearing the temporary database directory.